pom.xml

```diff
@@ -0,0 +1,574 @@
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.aliyun</groupId>
+  <artifactId>remote-shuffle-service</artifactId>
+  <version>0.3.0-SNAPSHOT</version>
+  <packaging>pom</packaging>
+
+  <name>Aliyun Shuffle Service Project Parent POM</name>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    <java.version>1.8</java.version>
+    <maven.compiler.source>${java.version}</maven.compiler.source>
+    <maven.compiler.target>${java.version}</maven.compiler.target>
+    <maven.version>3.5.4</maven.version>
+    <slf4j.version>1.7.16</slf4j.version>
+    <guava.version>14.0.1</guava.version>
+    <CodeCacheSize>512m</CodeCacheSize>
+    <codahale.metrics.version>3.2.6</codahale.metrics.version>
+    <javaxservlet.version>3.1.0</javaxservlet.version>
+    <!-- Apache Ratis version -->
+    <ratis.version>2.2.0</ratis.version>
+    <!-- ProtocolBuffer version, used to verify the protoc version and -->
+    <!-- define the protobuf JAR version                               -->
+    <protobuf.version>3.5.1</protobuf.version><!-- Maven protoc compiler -->
+    <protobuf-maven-plugin.version>0.5.1</protobuf-maven-plugin.version>
+    <!-- Allow modules to enable / disable certain build plugins easily. -->
+    <build.testJarPhase>prepare-package</build.testJarPhase>
+    <scalamock.version>5.1.0</scalamock.version>
+  </properties>
+
+  <modules>
+    <module>common</module>
+    <module>client</module>
+    <module>server-common</module>
+    <module>server-master</module>
+    <module>server-worker</module>
+  </modules>
+
+  <repositories>
+    <repository>
+      <!--
+        This is used as a fallback when the first try fails.
+      -->
+      <id>central</id>
+      <name>Maven Repository</name>
+      <url>https://repo.maven.apache.org/maven2</url>
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </repository>
+  </repositories>
+
+  <pluginRepositories>
+    <pluginRepository>
+      <id>central</id>
+      <url>https://repo.maven.apache.org/maven2</url>
+      <releases>
+        <enabled>true</enabled>
+      </releases>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </pluginRepository>
+  </pluginRepositories>
+
+  <dependencyManagement>
+    <dependencies>
```

Comment:
Currently, we have listed all the dependencies in advance, why not add them before using?

Reply:
Versions and exclusions have been carefully tuned, so I think it's better to inherit.